synapse/tests/test_tools_dmon.py

```diff
@@ -56,15 +56,28 @@ def test_main_lsboot(self):
         out0 = self.getTestOutp()
         out1 = self.getTestOutp()
 
-        s_dmon.main(['--log-level', 'debug', '--lsboot'], out0)
-        s_dmon.main(['--noboot', tfile.name], out1)
-
-        tfile.close()
+        try:
```

Comment:
Would it be better to just sleep for a while and retry the test if the box is OOM? Say try up to 3 times, sleep 10 between them, fail if it didnt pass after 3 tries? I'd just hate to have a test that should fail be skipped because of temporary conditions on the build box.

Reply:
Until we resolve these problems, maybe we create a decorator for SynTest that will retry a test a given number of times, with a given about of time to wait after each try, and only handles the OOM exception and raises whatever exception it got if it doesn't succeed after all its tries are used (I don't want this to catch any other exception)

Reply:
Definitely want to avoid non-deterministic tests.  What circumstances were causing the allocation failures?

Reply:
Hehe, also "try it a few times" is not a valid testing approach ;)

Reply:
LOL this branch should be killed.  it was due to cruddy CI issues.